include/curl/curl.h

```diff
@@ -748,6 +748,31 @@ typedef CURLcode (*curl_ssl_ctx_callback)(CURL *curl,    /* easy handle */
                                                           mbedtls_ssl_config */
                                           void *userptr);
 
+struct ssl_session_dump {
+  int backend;             /* curl_sslbackend enum value */
+  const char *scheme;      /* protocol scheme used */
+  const char *hostname;    /* host name for which this ID was used */
+  int port;                /* remote port */
+  unsigned char *sessionblob; /* serialized session data */
+  size_t blobsize;         /* session data size */
+};
+
+typedef CURLcode (*curl_ssl_cache_callback)(CURL *curl,
```

Comment:
I think `curl_ssl_cache_callback` is a bit too generic as it doesn't say what cache and it doesn't say what direction the data is going. How about `curl_sessionid_pull` or something? Ie the library pulls session ids from the application?

Reply:
I _think_ the others have "callback" particle in their name so for consistency it looks like it should be included here as well.
Perhaps something like `curl_sessionid_pull_callback()` or `curl_sessionid_load_callback()` just to avoid the use of yet another term beside the existing "import" and "load"?

Reply:
How about using read/write like we do for data?

Reply:
Sounds good. So
`curl_sessionid_write_callback()`
`curl_ssl_sessionid_write_callback()`
`curl_ssl_sess_write_callback()`
(and similar "read" counterparts).
You pick the one you like best. The last one is less explicit, but would avoid confusion when actually dealing with tickets.